### PR TITLE
Key pair generation and contract rewrite

### PR DESCRIPTION
**Pull request by Fantaskink**
- feat: Receive public key on election start and store in the contract
- feat: Rewrite smart contract to store encrypted votes in a single array.
- feat: Update vote routes.
- test: Add tests for vote routes.
- test: Modify contract tests

### CODE REVIEW

File: test/contracts/Election.test.js [hunk 1]
```diff
@@ -47,10 +56,31 @@ describe('Election Contract', function () {
         expect(await election.phase()).to.equal(ElectionPhase.Tallying); // Tallying phase
     });
 
+    it('Should allow uploading the encryption key in the registration phase', async function () {
+        await election.uploadEncryptionKey('key');
+        expect(await election.encryptionKey()).to.equal('key');
+    });
+
+    it('Should allow uploading the decryption key in the tallying phase', async function () {
```
Comment (MortenMunk) [inline test/contracts/Election.test.js:62]: This test does not assert anything

**Review by MortenMunk — CHANGES_REQUESTED** [review]

**Review by TheMOHD3886 — DISMISSED** [review]
lgtm 👍

**Review by Jatewo — APPROVED** [review]
@Jatewo Approved 👍

**Review by MortenMunk — APPROVED** [review]
lgtm

**Pull request merged**